internal/rest/http_client_test.go

```diff
@@ -0,0 +1,195 @@
+package rest
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+// roundTripFunc to mock and assert server side.
+type roundTripFunc func(req *http.Request) *http.Response
+
+func (f roundTripFunc) RoundTrip(req *http.Request) (*http.Response, error) {
+	return f(req), nil
+}
+
+//NewTestClient returns *http.Client with Transport replaced to avoid making real calls
+func setTransport(fn roundTripFunc, client *HTTPClient) {
+	client.httpClient.Transport = fn
+}
+
+func TestHTTPClient_Get(t *testing.T) {
+	type args struct {
+		serverHandler roundTripFunc
+		url           string
+		headers       []HTTPHeader
+	}
+	tests := []struct {
+		err  *Error
+		args args
+		name string
+		want []byte
```

Comment:
we don't use this "want" I guess.

Reply:
Thank you! Removed at https://github.com/hazelcast/hazelcast-go-client/pull/703/commits/48594298688387b76ed9fd3280296350ca7355fb